cmd/evm/internal/t8ntool/execution.go

```diff
@@ -441,33 +446,37 @@ func MakePreState(db ethdb.Database, chainConfig *params.ChainConfig, pre *Prest
 		}
 	}
 
-	// If verkle mode started, establish the conversion
+	// If verkle mode started
 	if verkle {
 		// Commit db an create a snapshot from it.
-		mptRoot, err := statedb.Commit(0, false)
+		treeRoot, err := statedb.Commit(0, false)
 		if err != nil {
 			panic(err)
 		}
-		rawdb.WritePreimages(mptSdb.DiskDB(), statedb.Preimages())
-		mptSdb.TrieDB().WritePreimages()
-		snaps, err := snapshot.New(snapshot.Config{AsyncBuild: false, CacheSize: 10}, mptSdb.DiskDB(), mptSdb.TrieDB(), mptRoot)
+
+		// If the active tree is a Verkle one, we're done.
+		if _, ok := statedb.GetTrie().(*trie.VerkleTrie); ok {
+			return statedb
+		}
+		// If isn't the case, configurate the conversion.
+
+		rawdb.WritePreimages(sdb.DiskDB(), statedb.Preimages())
+		sdb.TrieDB().WritePreimages()
+		snaps, err := snapshot.New(snapshot.Config{AsyncBuild: false, CacheSize: 10}, sdb.DiskDB(), sdb.TrieDB(), treeRoot)
 		if err != nil {
 			panic(err)
 		}
 		if snaps == nil {
 			panic("snapshot is nil")
 		}
-		snaps.Cap(mptRoot, 0)
-
-		// reuse the backend db so that the snapshot can be enumerated
-		sdb := mptSdb // := state.NewDatabaseWithConfig(db, &trie.Config{Verkle: true})
+		snaps.Cap(treeRoot, 0)
 
-		// Load the conversion status
+		// t loggggLoad the conversion status
 		log.Info("loading conversion status", "started", pre.Env.Started)
 		if pre.Env.Started != nil {
 			log.Info("non-nil started", "started", *pre.Env.Started)
 		}
-		sdb.InitTransitionStatus(pre.Env.Started != nil && *pre.Env.Started, pre.Env.Ended != nil && *pre.Env.Ended, mptRoot)
+		sdb.InitTransitionStatus(pre.Env.Started != nil && *pre.Env.Started, pre.Env.Ended != nil && *pre.Env.Ended, treeRoot)
```

Comment:
do we still need this one, though? I can't check because the test framework doesn't work on my machine, but it seems to be that this is already activated on line 433

Reply:
Makes sense to me. I'll remove it.